File: social-service/src/main/java/com/eirsteir/coffeewithme/social/domain/CreatedUpdatedDateTimeBaseModel.java
package com.eirsteir.coffeewithme.social.domain;

import java.util.Date;
import javax.persistence.MappedSuperclass;
import lombok.Getter;
import lombok.ToString;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Getter
@ToString
@SuperBuilder
@MappedSuperclass
public abstract class CreatedUpdatedDateTimeBaseModel {

  @CreationTimestamp private Date createdDateTime;

  @UpdateTimestamp private Date updatedDateTime;

  public CreatedUpdatedDateTimeBaseModel() {}
}

File: social-service/src/main/kotlin/com/eirsteir/coffeewithme/social/domain/CreatedUpdatedDateTimeBaseModel.kt
package com.eirsteir.coffeewithme.social.domain

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.util.*
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class CreatedUpdatedDateTimeBaseModel(
    @CreationTimestamp
    val createdDateTime: Date? = null,
    @UpdateTimestamp
    val updatedDateTime: Date? = null,
)